src/test/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistryTest.java

```diff
@@ -182,6 +183,18 @@ public void testGetArchiveRepoSpec() throws Exception {
         "  },",
         "  \"patch_strip\": 3",
         "}");
+    server.serve(
+        "/modules/baz/3.0/source.json",
+        "{",
+        "  \"url\": \"https://example.com/archive.jar?with=query\",",
+        "  \"integrity\": \"sha256-bleh\",",
+        "  \"overlay\": {",
+        "    \"BUILD.bazel\": {",
+        "       \"urls\": [\"http://mirror1\", \"http://mirror2\"],",
+        "       \"integrity\": \"sha256-bleh-overlay\"",
+        "   }",
+        "  }",
+        "}");
```

Comment:
I chose a structure (overlay) slightly different than the http_archive to simplify it.
http_archive is restricted to certain attr type (I.e str_dict_list) but in JSON we can be a lot more ergonomic to combine the integrity and the mirror URLS.

Reply:
just driving by -- I think this is a great idea!